app/src/main/java/com/example/bigowlapp/repository/NotificationRepository.java

```diff
@@ -1,11 +1,52 @@
 package com.example.bigowlapp.repository;
 
+import androidx.annotation.VisibleForTesting;
+
 import com.example.bigowlapp.model.Notification;
+import com.example.bigowlapp.model.SupervisionRequest;
+import com.example.bigowlapp.utils.Constants;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+import com.google.firebase.firestore.QuerySnapshot;
+
+import java.util.ArrayList;
+import java.util.List;
 
 public class NotificationRepository extends Repository<Notification> {
 
     // TODO: Dependency Injection Implementation for Firestore
     public NotificationRepository() {
         super("notifications");
     }
+
+    // TODO: Remove or Modify when dependency injection implemented
+    @VisibleForTesting
```

Comment:
Aren't all methods visible for testing? Why do you need this?

Reply:
Its just to specify that the function should only be used for testing, and no where else. That is to say, that the IDE can use the annotation to stop someone from using it for non-testing code.

https://developer.android.com/reference/androidx/annotation/VisibleForTesting